src/semver/version.py

```diff
@@ -661,6 +661,35 @@ def isvalid(cls, version: str) -> bool:
         except ValueError:
             return False
 
+    def is_compatible(self, callee: 'Version') -> bool:
+        """
+        Check if caller is compatible with callee.
+
+        The algorithm checks:
+
+        * if the two majors are different, it's incompatible
+        * if the two majors are equal, but the minor of the callee
+          is lower than the caller, it's incompatible
+        * if both prereleases are present and different,
+          it's incompatible
+        * otherwise it's compatible
+
+        :param callee: the callee to check for compatibility
+        :return: True, if caller is compatible callee,
+                 otherwise False
+        """
+        if not isinstance(callee, type(self)):
+            raise TypeError(
+                f"Expected a Version type but got {type(callee)}"
+            )
+        if self._major != callee.major:
```

Comment:
✍️ `elif`?

Reply:
I think, in that case it doesn't matter. Python won't continue after a `raise`. :wink: